plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -0,0 +1,657 @@
+	 *
+	 * @param string $order The unsanitized order.
+	 * @return string The sanitized order.
+	 */
+	private function sanitize_order( string $order ): string {
+		$order = strtoupper( is_string( $order ) ? $order : '' );
+
+		return in_array( $order, array( 'ASC', 'DESC' ), true ) ? $order : 'DESC';
+	}
+
+	/**
+	 * Builds the final SQL query to be run.
+	 *
+	 * @return void
+	 */
+	private function build_query(): void {
+		$this->maybe_remap_args();
+
+		// Build query.
+		$this->process_orders_table_query_args();
+		$this->process_operational_data_table_query_args();
+		$this->process_addresses_table_query_args();
+
+		// Meta queries.
+		if ( ! empty( $this->args['meta_query'] ) ) {
+			$this->meta_query = new OrdersTableMetaQuery( $this );
+
+			$sql = $this->meta_query->get_sql_clauses();
+
+			$this->join  = $sql['join'] ? array_merge( $this->join, $sql['join'] ) : $this->join;
+			$this->where = $sql['where'] ? array_merge( $this->where, array( $sql['where'] ) ) : $this->where;
+
+			if ( $sql['join'] ) {
+				$this->groupby[] = "{$this->tables['orders']}.id";
+			}
+		}
+
+		$this->process_orderby();
+		$this->process_limit();
+
+		$orders_table = $this->tables['orders'];
+
+		// DISTINCT.
+		$distinct = '';
+
+		// SELECT [fields].
+		$this->fields = "{$orders_table}.id";
+		$fields       = $this->fields;
+
+		// SQL_CALC_FOUND_ROWS.
+		if ( ( ! $this->arg_isset( 'no_found_rows' ) || ! $this->args['no_found_rows'] ) && $this->limits ) {
+			$found_rows = 'SQL_CALC_FOUND_ROWS';
+		} else {
+			$found_rows = '';
+		}
+
+		// JOIN.
+		$join = implode( ' ', $this->join );
+
+		// WHERE.
+		$where = '1=1';
+		foreach ( $this->where as $_where ) {
+			$where .= " AND ({$_where})";
+		}
+
+		// ORDER BY.
+		$orderby = $this->orderby ? ( 'ORDER BY ' . $this->orderby ) : '';
+
+		// LIMITS.
+		$limits = $this->limits ? 'LIMIT ' . implode( ',', $this->limits ) : '';
+
+		// GROUP BY.
+		$groupby = $this->groupby ? 'GROUP BY ' . implode( ', ', (array) $this->groupby ) : '';
+
+		$this->sql = "SELECT $found_rows $distinct $fields FROM $orders_table $join WHERE $where $groupby $orderby $limits";
+	}
+
+	/**
+	 * Generates a properly escaped and sanitized WHERE condition for a given field.
+	 *
+	 * @param string $table    The table the field belongs to.
+	 * @param string $field    The field or column name.
+	 * @param mixed  $value    The value.
+	 * @param string $type     The column type as specified in {@see OrdersTableDataStore} column mappings.
+	 * @param string $operator The operator to use in the condition. Defaults to '=' or 'IN' depending on $value.
+	 * @return string The resulting WHERE condition.
+	 */
+	public function get_where_condition( string $table, string $field, $value, string $type, string $operator = '' ): string {
+		global $wpdb;
+
+		$db_util  = wc_get_container()->get( DatabaseUtil::class );
+		$operator = strtoupper( $operator ? $operator : '=' );
```

Comment:
we can make this more explicit:
```suggestion
		$operator = strtoupper( '' !== $operator ? $operator : '=' );
```

Reply:
Done.